StateMachine: Merge Commit and Transfer (#83)
tigerbeetle/tigerbeetle: docs/DESIGN.md

**create_account**: Create an account.

* We use the terms `credit` and `debit` instead of "payable" or "receivable" since the meaning of a credit balance depends on whether the account is an asset or liability or equity, income or expense.

> * An `posted` amount refers to an amount posted by a committed transfer.

Reviewer: ```suggestion
* A `posted` amount refers to an amount posted by a transfer.
```

> * A `pending` amount refers to an inflight amount posted by a two-phace commit transfer only, where the commit is still outstanding, and where the transfer timeout has not yet fired. In other words, the transfer amount has been reserved in the account balance (to avoid double-spending) but not yet committed. The reserved amount will rollback if the transfer ultimately fails.

Reviewer: ```suggestion
* A `pending` amount refers to an inflight amount yet-to-be-posted by a two-phase commit transfer only, where the commit is still outstanding, and where the transfer timeout has not yet fired. In other words, the transfer amount has been reserved in the account balance (to avoid double-spending) but not yet committed. The reserved amount will rollback if the transfer ultimately fails. By default, transfers post automatically, but being able to reserve the amount as pending and then post the amount only later can sometimes be convenient, for example when switching credit card payments.
```

> * The total debit balance of an account is given by adding `debits_posted` plus `debits_pending`. Likewise for the total credit balance of an account.

Reviewer: ```suggestion
* The debit balance of an account is given by adding `debits_posted` plus `debits_pending`. Likewise for the credit balance of an account.
```

> * A `pending` amount refers to an inflight amount yet-to-be-posted by a two-phase commit transfer only, where the commit is still outstanding, and where the transfer timeout has not yet fired. In other words, the transfer amount has been reserved in the account balance (to avoid double-spending) but not yet committed. The reserved amount will rollback if the transfer ultimately fails. By default, transfers post automatically, but being able to reserve the amount as pending and then post the amount only later can sometimes be convenient, for example when switching credit card payments.

Reviewer: ```suggestion
* A `pending` amount refers to an inflight amount yet-to-be-posted by a two-phase transfer only, where the transfer is still pending, and where the transfer timeout has not yet fired. In other words, the transfer amount has been reserved in the pending account balance (to avoid double-spending) but not yet posted to the posted balance. The reserved amount will rollback if the transfer ultimately fails. By default, transfers post automatically, but being able to reserve the amount as pending and then post the amount only later can sometimes be convenient, for example when switching credit card payments.
```
